deploy/charts/cert-manager/templates/rbac.yaml

```diff
@@ -10,14 +10,19 @@ metadata:
     heritage: {{ .Release.Service }}
 rules:
   - apiGroups: ["certmanager.k8s.io"]
-    resources: ["certificates", "certificates/finalizers", "issuers", "clusterissuers", "orders", "orders/finalizers", "challenges"]
+    resources: ["certificates", "certificates/finalizers", "issuers", "issuers/finalizers","clusterissuers", "orders", "orders/finalizers", "challenges",  "challenges/finalizers"]
     verbs: ["*"]
   - apiGroups: [""]
-    resources: ["configmaps", "secrets", "events", "services", "pods"]
+    resources: ["configmaps", "secrets", "secrets/finalizers","events", "services", "services/finalizers","pods", "pods/finalizers"]
     verbs: ["*"]
   - apiGroups: ["extensions"]
-    resources: ["ingresses"]
+    resources: ["ingresses", "ingresses/finalizers", "ingresses/status"]
     verbs: ["*"]
+{{- if .Values.global.isOpenshift }}
+  - apiGroups: ["route.openshift.io"]
+    resources: ["routes", "routes/custom-host", "routes/finalizers"]
```

Comment:
Is this required? We don't modify 'route' resources ever, so I'm surprised this is needed 🤔

Reply:
So, i will first answer this , as this was the most difficult one to find.
i have no idea why , but it looks like openshift NEEDS these rules to modify the ingress.